src/main/java/frc/robot/subsystems/climber/ClimberIOTalonFX.java

```diff
@@ -0,0 +1,225 @@
+    rightStatorCurrentAmpsStatusSignal = rightMotor.getStatorCurrent();
+    rightSetpointStatusSignal = rightMotor.getClosedLoopReference();
+
+    leftPositionCurrentRequest = new MotionMagicExpoTorqueCurrentFOC(0.0);
+    rightPositionCurrentRequest = new MotionMagicExpoTorqueCurrentFOC(0.0);
+  }
+
+  /**
+   * Update the inputs based on the current state of the TalonFX leftMotor controller.
+   *
+   * @param inputs the inputs object to update
+   */
+  @Override
+  public void updateInputs(ClimberIOInputs inputs) { // FIXME: May need inputs for both motors
+    BaseStatusSignal.refreshAll(
+        leftVelocityRPSStatusSignal,
+        leftPositionRotationsStatusSignal,
+        leftStatorCurrentAmpsStatusSignal,
+        leftSetpointStatusSignal,
+        leftSupplyCurrentAmpsStatusSignal,
+        rightVelocityRPSStatusSignal,
+        rightPositionMetersStatusSignal,
+        rightStatorCurrentAmpsStatusSignal,
+        rightSetpointStatusSignal,
+        rightSupplyCurrentAmpsStatusSignal);
+
+    inputs.leftVelocityRPS = leftVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.leftPositionMeters =
+        leftPositionRotationsStatusSignal.getValueAsDouble() * DRUM_CIRCUMFERENCE;
+    inputs.leftStatorCurrentAmps = leftStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSupplyCurrentAmps = leftSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.leftSetpoint = leftSetpointStatusSignal.getValueAsDouble();
+
+    inputs.rightVelocityRPS = rightVelocityRPSStatusSignal.getValueAsDouble();
+    inputs.rightPositionMeters = rightPositionMetersStatusSignal.getValueAsDouble();
+    inputs.rightStatorCurrentAmps = rightStatorCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSupplyCurrentAmps = rightSupplyCurrentAmpsStatusSignal.getValueAsDouble();
+    inputs.rightSetpoint = rightSetpointStatusSignal.getValueAsDouble();
+
+    // inputs.closedLoopError = leftMotor.getClosedLoopError().getValue();
+    // inputs.power =
+    // inputs.controlMode = leftMotor.getControlMode().toString();
+
+    // update configuration if tunables have changed
+    if (kP.hasChanged()
+        || kI.hasChanged()
+        || kD.hasChanged()
+        || kS.hasChanged()
+        || kA.hasChanged()
+        || kV.hasChanged()
+        || kVExpo.hasChanged()
+        || kAExpo.hasChanged()) {
+      TalonFXConfiguration config = new TalonFXConfiguration();
+      MotionMagicConfigs motionMagicConfigs = new MotionMagicConfigs();
+
+      this.leftMotor.getConfigurator().refresh(config);
+      config.Slot0.kP = kP.get();
+      config.Slot0.kI = kI.get();
+      config.Slot0.kD = kD.get();
+      config.Slot0.kS = kS.get();
+      config.Slot0.kV = kV.get();
+      config.Slot0.kA = kA.get();
+      motionMagicConfigs.MotionMagicExpo_kV = kVExpo.get();
+      motionMagicConfigs.MotionMagicExpo_kA = kAExpo.get();
+
+      this.leftMotor.getConfigurator().apply(config);
+      this.rightMotor.getConfigurator().apply(config);
```

Comment:
We need to refresh the config and reset all the values before applying:
```
this.rightMotor.getConfigurator().refresh(config);
```

Otherwise, we will change other settings (e.g., inverted motor)on the right motor with the value from the left motor.